pkg/dio/multi.go

```diff
@@ -192,6 +192,11 @@ func (t *multiReader) Read(p []byte) (int, error) {
 		}
 	}
 
+	// return 0, nil if there are no readers
+	if len(t.readers) == 0 {
```

Comment:
On the detach, the SSH connection sends a EOF and drops the connection, since t.err is io.EOF on that case we don't block but we also don't want to propagate that error to the caller hence we swallow it